smalltalksrc/VMMaker/DruidJIT.class.st

```diff
@@ -5153,79 +5149,80 @@ DruidJIT >> gen_PrimitiveNewWithArg [
 	self MoveR: Extra3Reg Mw: 16 r: SPReg.
 	self MoveR: Extra0Reg R: Extra3Reg.
 	self MoveR: Extra3Reg Mw: 24 r: SPReg.
+	self MoveR: Extra2Reg R: Extra1Reg.
 	currentBlock := self Label.
 	jump7 jmpTarget: currentBlock.
 	self MoveMw: 0 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: SendNumArgsReg.
+	self MoveR: Extra3Reg R: Extra2Reg.
 	s172 := 0.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: ClassReg.
-	self CmpCq: 16rFFFF R: ClassReg.
+	self MoveR: Extra3Reg R: SendNumArgsReg.
+	self CmpCq: 16rFFFF R: SendNumArgsReg.
 	jumpTrue := self JumpAbove: 0.
-	self MoveCq: 0 R: ClassReg.
+	self MoveCq: 0 R: SendNumArgsReg.
 	jumpNext := self Jump: 0.
 	jumpTrue jmpTarget: self Label.
-	self MoveCq: 1 R: ClassReg.
+	self MoveCq: 1 R: SendNumArgsReg.
 	jumpNext jmpTarget: self Label.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self CmpCq: 16rFFFF R: Extra3Reg.
 	jump7 := self JumpBelowOrEqual: 0.
 	jump10 := self Jump: 0.
```

Comment:
We could also have some code layout algorithms and some heuristics of common paths

Reply:
https://github.com/Alamvic/druid/issues/106